### ISSUE TITLE
slow response from file checking system

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1438]
It looks like this file:
https://dev.api.cottagelabs.com/service/oab/permissions?url=https://github.com/OAButton/discussion/files/4251502/revised.manuscript.December.2019.docx&doi=10.1007/s10508-020-01626-w

is really really really really slow in the file checking system, to the point that I think sometimes it just fails. 

Can be tested on the UI using:

https://openaccessbutton.org/deposit?doi=10.1007/s10508-020-01626-w
[revised manuscript December 2019.docx](https://github.com/OAButton/discussion/files/4251502/revised.manuscript.December.2019.docx)

**Comment by markmacgillivray** [comment]
Problem found, solution implemented. Testing ongoing, will be ready to go live on Monday, along with push to live of other things that are completed.

**Comment by markmacgillivray** [comment]
There had been an update to the lib we use to extract text from docx documents (textract, with antiword under it). It appeared to work better than the old one, and I didn't see any problems testing it so I put in that upgrade at some point in the last few months when we did a push. However that file, and some others I have tried, definitely take a very long time when passed to textract. The method it runs to identify the mimetype from the file name seemed perhaps to be slow so I added mime guessing before running with textract but even then it was slow. So I got another lib to do the job for docx (mammoth). That has been installed and tested directly, and returns much faster, back to the speeds we expect and see for other extractions. However an error is thrown when that new method is called from permissions/deposit, so just finishing testing of that to see exactly what values I need to pass to get it to run smooth.

The switch to a different lib requires an update of the dependencies as well as the API code. That can all get done on Monday, won't take too long as everything else is already up to date.
